processing/src/main/java/io/druid/query/extraction/MatchingDimExtractionFn.java

```diff
@@ -77,6 +77,12 @@ public boolean preservesOrdering()
   }
 
   @Override
+  public ExtractionType getExtractionType()
+  {
+    return ExtractionType.ONE_TO_ONE;
```

Comment:
Technically speaking, this can be a MANY_TO_ONE mapping things that don't match to `null`